smithy-model/src/main/java/software/amazon/smithy/model/validation/suppressions/ModelBasedEventDecorator.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.model.validation.suppressions;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.SourceException;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.SuppressTrait;
+import software.amazon.smithy.model.validation.Severity;
+import software.amazon.smithy.model.validation.ValidatedResult;
+import software.amazon.smithy.model.validation.ValidationEvent;
+import software.amazon.smithy.model.validation.ValidationEventDecorator;
+import software.amazon.smithy.utils.SmithyUnstableApi;
+
+/**
+ * Creates a {@link ValidationEventDecorator} that applies custom suppressions, custom severity overrides,
+ * suppressions parsed from model metadata, and severity overrides parsed from model metadata.
+ */
+@SmithyUnstableApi
+public final class ModelBasedEventDecorator {
+
+    private static final String SUPPRESSIONS = "suppressions";
+    private static final String SEVERITY_OVERRIDES = "severityOverrides";
+
+    private final List<Suppression> suppressions = new ArrayList<>();
+    private final List<SeverityOverride> severityOverrides = new ArrayList<>();
+
+    /**
+     * Sets custom suppressions not found in the model.
+     *
+     * @param suppressions Suppressions to set.
+     * @return Returns the ModelBasedEventDecorator.
+     */
+    public ModelBasedEventDecorator suppressions(Collection<? extends Suppression> suppressions) {
```

Comment:
This one is not use at the moment, is the idea to keep it around as a possible extension point in the future or can we remove it for now?